File: color_segmentation/hsv_segmentation.py
import cv2
import numpy as np

cv2.samples.addSamplesDataSearchPath("../test_images")

fruits = cv2.imread(cv2.samples.findFile('fruits.jpg'))
fruits_hsv = cv2.cvtColor(fruits, cv2.COLOR_BGR2HSV)

cv2.namedWindow("Image")

def do_nothing(value):
    pass

cv2.createTrackbar("low_h", "Image", 0, 179, do_nothing)
cv2.createTrackbar("low_s", "Image", 0, 255, do_nothing)
cv2.createTrackbar("low_v", "Image", 0, 255, do_nothing)

cv2.createTrackbar("high_h", "Image", 179, 179, do_nothing)
cv2.createTrackbar("high_s", "Image", 255, 255, do_nothing)
cv2.createTrackbar("high_v", "Image", 255, 255, do_nothing)

while True:
    low_h = cv2.getTrackbarPos("low_h", "Image")
    low_s = cv2.getTrackbarPos("low_s", "Image")
    low_v = cv2.getTrackbarPos("low_v", "Image")

    high_h = cv2.getTrackbarPos("high_h", "Image")
    high_s = cv2.getTrackbarPos("high_s", "Image")
    high_v = cv2.getTrackbarPos("high_v", "Image")

    mask = cv2.inRange(fruits_hsv, np.array([low_h, low_s, low_v]), np.array([high_h, high_s, high_v]))

    segmented_fruits = cv2.bitwise_and(fruits, fruits, mask=mask)
    cv2.imshow("Image", segmented_fruits)
    if cv2.waitKey(1) & 0xFF == ord('q'):
        break

File: cpp/color_segmentation/HSVSegmentation.cpp
#include <opencv2/opencv.hpp>

int main()
{
    cv::samples::addSamplesDataSearchPath("../../test_images");
    cv::String img_path = cv::samples::findFile("fruits.jpg");
    cv::Mat img = cv::imread(img_path), img_hsv;
    cv::cvtColor(img, img_hsv, cv::COLOR_BGR2HSV);

    cv::namedWindow("Image");

    int low_h = 0, low_s = 0, low_v = 0;
    int high_h = 179, high_s = 255, high_v = 255;

    cv::createTrackbar("low_h", "Image", &low_h, 179);
    cv::createTrackbar("low_s", "Image", &low_s, 255);
    cv::createTrackbar("low_v", "Image", &low_v, 255);
    cv::createTrackbar("high_h", "Image", &high_h, 179);
    cv::createTrackbar("high_s", "Image", &high_s, 255);
    cv::createTrackbar("high_v", "Image", &high_v, 255);

    while (1)
    {
        std::vector<int> lowhsv = {low_h, low_s, low_v};
        std::vector<int> highhsv = {high_h, high_s, high_v};
        cv::Mat mask, segmented_fruits;

        cv::inRange(img_hsv, lowhsv, highhsv, mask);

        cv::bitwise_and(img, img, segmented_fruits, mask);
        cv::imshow("Image", segmented_fruits);

        if (cv::waitKey(1) == 'q')
            break;
    }
    return 0;
}